application/pages/WorldListPage.cpp

```diff
@@ -0,0 +1,94 @@
+//
+// Created by robotbrain on 8/18/15.
+//
+
+#include "WorldListPage.h"
+#include "ui_WorldListPage.h"
+#include "minecraft/WorldList.h"
+#include "dialogs/ModEditDialogCommon.h"
+#include <QEvent>
+#include <QKeyEvent>
+
+WorldListPage::WorldListPage(BaseInstance *inst, std::shared_ptr<WorldList> worlds, QString id,
+							 QString iconName, QString displayName, QString helpPage,
+							 QWidget *parent)
+	: QWidget(parent), ui(new Ui::WorldListPage)
+{
+	ui->setupUi(this);
+	ui->tabWidget->tabBar()->hide();
+	m_inst = inst;
+	m_worlds = worlds;
+	m_id = id;
+	m_displayName = displayName;
+	m_iconName = iconName;
+	m_helpName = helpPage;
+	ui->worldTreeView->setModel(m_worlds.get());
+	ui->worldTreeView->installEventFilter(this);
+	auto smodel = ui->worldTreeView->selectionModel();
+	connect(smodel, SIGNAL(currentChanged(QModelIndex, QModelIndex)),
+			SLOT(modCurrent(QModelIndex, QModelIndex)));
+}
+
+void WorldListPage::opened()
+{
+	m_worlds->startWatching();
+}
+
+void WorldListPage::closed()
+{
+	m_worlds->stopWatching();
+}
+
+WorldListPage::~WorldListPage()
+{
+	m_worlds->stopWatching();
+	delete ui;
+}
+
+bool WorldListPage::shouldDisplay() const
+{
+	if (m_inst)
+		return !m_inst->isRunning();
+	return true;
+}
+
+bool WorldListPage::worldListFilter(QKeyEvent *keyEvent)
```

Comment:
With just a few more things I'd still change these. You can't tell at a quick glance what these functions are supposed to do (which should be possible since what they're doing is quite trivial).